india_forecast_app/models/pvnet/utils.py

```diff
@@ -134,12 +144,24 @@ def process_and_cache_nwp(source_nwp_path: str, dest_nwp_path: str):
         ds.__setitem__("variable", new_variables)
 
     # Hack to resolve some NWP data format differences between providers
-    elif is_gfs:
+    elif nwp_config.source == "gfs":
         data_var = ds[list(ds.data_vars.keys())[0]]
         # # Use .to_dataset() to split the data variable based on 'variable' dim
         ds = data_var.to_dataset(dim="variable")
         ds = ds.rename({"t2m": "t"})
+
+    if nwp_config.source == "mo_global":
+
+        # only select the variables we need
+        ds = ds.sel(variable=["temperature_sl", "wind_u_component_10m", "wind_v_component_10m"])
```

Comment:
Hardcoding the variables here seems a bit risky, the weather variable selection is handled later on by datapipes/data-sampler so not sure if it is worth having this hardcoded here as well as there would be different variables for PV or wind

Reply:
yea good point. I can take them from the config though.